docs/mana_proposal.md

```diff
@@ -0,0 +1,100 @@
+# Mana Implementation Proposal
+
+The goal of this document is to provide a high level overview of how mana will be implemented in GoShimmer.
+
+## Introduction
+
+Mana is a a reputation system for nodes within the IOTA network.
+
+Reputation is gained by contributing to the network, i.e. creating value transfers.
+Reputation is lost when not cooperating with the network: using up all the available bandwidth, advocating for
+"bad" transfers during voting, etc..
+
+As time passes, earned mana of a node decays to encourage keeping up the good behavior.
+
+## Scope
+
+The sope of the first implementation of mana into GoShimmer is to verify that mana calculations work,
+study mana distribution in the test network and verify that nodes have similar view on the network. A test dApp could
+be written to see how nodes react to "simulated misbehaving" nodes.
+
+## Mana Calculation
+
+Mana is esentially the reputation score of a node in the IOTA network. Mana is calculated locally in each node, as a
+function that takes value transactions as input and produces the Base Mana Vector as output.
+It consists of Base Mana 1 and Base Mana 2. Given a value transaction, these are determined as follows:
+ 1. Base Mana 1 is revoked from the node that created the output(s) used as input(s) in the transaction, and is pledged to
+    the node creating the new output(s).
+ 2. Base Mana 2 is freshly created at the issuance time of the transaction, awarded to the node, but decays with time.
+
+Access and Consensus Mana (mana used by other node modules) are derived from Base Mana 1 and 2 and "smoothened" through
+exponential moving averages. Since Base Mana 2 decays, it is important to find a suitable interval at which the value
+is updated in code. Same goes for Access and Consensus Mana,
+as EMAs are also time dependent.
+
+The exact mathematical formulas, and their respective parameters will be determined later.
+
+## Mana Plugin in GoShimmer
+
+Mana should be implemented as a separate plugin in GoShimmer.
+
+The `mana plugin` is responsible for:
+ - calculating mana from value transactions,
+ - keeping a log of the different mana values of all nodes,
+ - updating mana values in a timely fashion,
+ - respond to mana related queries from other modules,
+ - allowing certain modules to overwrite mana of nodes (punishing bad behavior).
+
+ The proposed mana plugin in should keep track of the different mana values of nodes and handle calculation and timely
+ updates. Mana values are mapped to `nodeID`s and stored in a `map` data structure:
+ ```go
+baseMana1Map := map[<nodeID>[]byte]<baseMana1-value>float64, and so on...
+```
+ Base Mana 2, EMAs, Consensus and Access Mana records are updated every `t` interval with tickers in a dedicated `BackgrounWorker`.
```

Comment:
From mana spec
>The Base Mana Snapshot State will only be updated when transactions are snapshotted
This is, therefore, a function of `snapshot` instead of a timer.

Or are we sticking to a timer for this first iteration?

Reply:
Base Mana Snapshot State concerns Base Mana 1 and 2 values and the time these values were last updated, right? In the first iteration we will not have a Mana Snapshot State as it implies you have the message/transaction `snapshotting` in place, which we obviously don't. Anyway, even when we will have this, the decaying parts plus the EMAs should be updated at regular time intervals.
So, mana snapshot file will be updated upon `snapshotting`, but derived values will have to be updated regularly regardless of `snapshotting`. That is why we need timers/tickers/whatever.